hoomd/mpcd/ConcentricCylindersGeometry.h

```diff
@@ -0,0 +1,186 @@
+        const Scalar c = pos.x * pos.x + pos.y * pos.y - R * R;
+        dt = (-b + sign * (slow::sqrt(b * b - Scalar(4) * a * c))) / (Scalar(2) * a);
+
+        // backtrack the particle for dt to get to point of contact
+        pos -= vel * dt;
+
+        // update velocity according to boundary conditions
+        // no-slip requires reflection of the tangential components.
+        if (sign == 1)
+            {
+            // no-slip requires reflection of the tangential components.
+            if (m_no_slip)
+                {
+                vel = -vel;
+                }
+            // slip requires reflection of the normal components.
+            else
+                {
+                vel.x -= Scalar(2) * pos.x * pos.x * vel.x / (R * R);
+                vel.y -= Scalar(2) * pos.y * pos.y * vel.y / (R * R);
+                }
+            }
+
+        if (sign == -1)
+            {
+            if (m_no_slip)
+                {
+                vel.x = -vel.x + Scalar(2) * m_V * -pos.y;
+                vel.y = -vel.y + Scalar(2) * m_V * pos.x;
+                vel.z = -vel.z;
+                }
+            else
+                {
+                vel.x -= Scalar(2) * pos.x * pos.x * vel.x / (R * R);
+                vel.y -= Scalar(2) * pos.y * pos.y * vel.y / (R * R);
+                }
+            }
+
+        return true;
+        }
+
+    //! Check if a particle is out of bounds
+    /*!
+     * \param pos Current particle position
+     * \returns True if particle is out of bounds, and false otherwise
+     */
+    HOSTDEVICE bool isOutside(const Scalar3& pos) const
+        {
+        const Scalar r = pos.x * pos.x + pos.y * pos.y;
+        return (r > m_R12 || r < m_R02);
+        }
+
+    //! Get inner radius of the cylinder
+    /*!
+     * \returns Inner radius of the cylinder
+     */
+    HOSTDEVICE Scalar getR0() const
+        {
+        return m_R0;
+        }
+
+    //! Get outer radius of the cylinder
+    /*!
+     * \returns Outer radius of the cylinder
+     */
+    HOSTDEVICE Scalar getR1() const
+        {
+        return m_R1;
+        }
+
+    //! Get the speed of the outer cylinder
+    /*!
+     * \returns Speed of the outer cylinder
+     */
+    HOSTDEVICE Scalar getSpeed() const
+        {
+        return m_V;
+        }
+
+    //! Get the wall boundary condition
+    /*!
+     * \returns Boundary condition at wall
+     */
+    HOSTDEVICE bool getNoSlip() const
+        {
+        return m_no_slip;
+        }
+
+#ifndef __HIPCC__
+    //! Get the unique name of this geometry
+    static std::string getName()
+        {
+        return std::string("ConcentricCylinders");
+        }
+#endif // __HIPCC__
+
+    private:
+    const Scalar m_R0;    //!< Inner radius of the cylinder
+    const Scalar m_R02;   //!< Square of inner radius
+    const Scalar m_R1;    //!< Outer radius of the cylinder
+    const Scalar m_R12;   //!< Square of outer radius
```

Comment:
I think you only ever actually used the squared versions of these, so you can just save those. Have the getters return the sqrt of these.
```suggestion
    const Scalar m_R0_sq;   //!< Square of inner radius
    const Scalar m_R1_sq;   //!< Square of outer radius
```